dbux-code/src/projectView/projectViewController.js

```diff
@@ -27,14 +28,57 @@ const externals = {
 
 class ProjectViewController {
   constructor(context) {
-    this.treeDataProvider = new ProjectNodeProvider(context, this);
-    this.treeView = this.treeDataProvider.treeView;
-
     // ########################################
     //  init projectManager
     // ########################################
     this.manager = initDbuxProjects(cfg, externals);
     debug(`Initialized dbux-projects. Projects folder = "${path.resolve(cfg.projectsRoot)}"`);
+
+    // ########################################
+    //  init treeView
+    // ########################################
+    this.treeDataProvider = new ProjectNodeProvider(context, this);
+    this.treeView = this.treeDataProvider.treeView;
+  }
+
+  async activateBugByNode(bugNode, debugMode = false) {
+    const options = {
+      title: `[dbux] Activating bug:{${bugNode.bug.name}}`
+    };
+
+    runTaskWithProgressBar(async (progress, cancelToken) => {
+      await this._activateBug(progress, cancelToken, bugNode, debugMode);
+    }, options);
+  }
+
+  async _activateBug(progress, cancelToken, bugNode, debugMode) {
+    const { bug } = bugNode;
+    const runner = this.manager.getOrCreateRunner();
+
+    // cancel any currently running tasks
+    await runner.cancel();
+    progress.report({ increment: 20, message: 'activating...' });
+
+    // activate/install project
+    if (cancelToken.isCancellationRequested) {
+      return;
+    }
+    await runner.activateProject(bugNode.bug.project);
+    progress.report({ increment: 40, message: 'opening in editor...' });
+
+    // open in editor (must be after activation/installation)
+    if (cancelToken.isCancellationRequested) {
+      return;
+    }
+    await bug.openInEditor();
```

Comment:
no need to open in editor here. we want a button for that instead. In fact we want two buttons:
1. "open project in editor" (add to workspace)
2. "open bug in editor" (show)